internal/controller/runtime/fsm/testing/fake_k8s_client_config.go

```diff
@@ -20,13 +21,47 @@ func GetFakePatchInterceptorFn(incGeneration bool) func(ctx context.Context, cli
 		if !ok {
 			return errors.New("failed to cast object to shoot")
 		}
-		if incGeneration {
+		if incShootGeneration {
 			shoot.Generation++
 		}
 		return nil
 	}
 }
 
+func GetFakePatchInterceptorForConfigMap(incShootGeneration bool) func(ctx context.Context, client client.WithWatch, obj client.Object, patch client.Patch, opts ...client.PatchOption) error {
+	return func(ctx context.Context, client client.WithWatch, obj client.Object, patch client.Patch, opts ...client.PatchOption) error {
+		// Apply patches are supposed to upsert, but fake client fails if the object doesn't exist,
+		// Update the generation to simulate the object being updated using interceptor function.
+		if patch.Type() != types.ApplyPatchType {
+			return client.Patch(ctx, obj, patch, opts...)
+		}
+		if incShootGeneration {
+			shoot, ok := obj.(*gardener_api.Shoot)
+			if ok {
+				shoot.Generation++
```

Comment:
1) The declaration is misleading. The parameter incShootGeneration should affect only operations on shoot. 
2) It can be written in simpler way - without else: 
```suggestion
				shoot.Generation++
				return nil
```